### PR TITLE
On citations count... again...

### PR DESCRIPTION
**Pull request by pazembrz**

### CODE REVIEW

**Comment by pazembrz** [comment]
ci test please

File: inspirehep/modules/records/api.py [hunk 1]
```diff
@@ -610,7 +610,13 @@ def _query_citing_records(self, show_duplicates=False, session=None):
         citation_query = session.query(RecordMetadata).with_entities(RecordMetadata.id,
                                                                      RecordMetadata.json['control_number'])
         citation_filter = referenced_records(RecordMetadata.json).contains([ref])
-        citations = citation_query.filter(citation_filter)
+        filter_deleted_records = or_(not_(type_coerce(RecordMetadata.json, JSONB).has_key('deleted')),  # noqa: W601
+                                     not_(RecordMetadata.json['deleted'] == cast(True, JSONB)))
+        only_literature_collection = type_coerce(RecordMetadata.json['_collections'],
```
Comment (michamos) [inline inspirehep/modules/records/api.py]: you can do this in a simpler way:
https://github.com/inspirehep/inspire-next/blob/4d116e13e625dc4ea4a8b9a3ab2bcf6261ecec77/inspirehep/modules/disambiguation/core/db/readers.py#L85-L86

File: inspirehep/modules/search/api.py [hunk 2]
```diff
@@ -99,21 +91,11 @@ def mget(self, uuids, **kwargs):
...
-            user_coll = user_collections
-
-        query = Q('match', _collections=collection)
-
-        for collection in list(all_restricted_collections - user_coll):
-            query = query & ~Q('match', _collections=collection)
-
-        return query
+        collection = request.values.get('cc', "Literature")
+    query = Q('match', _collections=collection) & Q('bool', filter=[~Q('exists', field='deleted')])
```
Comment (michamos) [inline inspirehep/modules/search/api.py]: you don't need the second filter as deleted records are not indexed

**Comment by pazembrz** [comment]
ci test please

**Comment by pazembrz** [comment]
ci test please

**Comment by pazembrz** [comment]
ci merge please

**Pull request merged**